crates/sparrow-interfaces/src/source.rs

```diff
@@ -0,0 +1,12 @@
+use arrow_schema::SchemaRef;
+use futures::stream::BoxStream;
+use sparrow_batch::Batch;
+use sparrow_core::ReadConfig;
```

Comment:
If `ReadConfig` was here, then it wouldn't need to depend on `sparrow_core`, right?